addon/templates/components/frost-list-item-content.hbs

```diff
@@ -0,0 +1,67 @@
+<div class={{concat
+    'frost-list-item-container'
+    (if (eq index 0) ' first')
+    (if (eq index (sub _items.length 1)) ' last')
+    (if model.states.isSelected ' is-selected')
+  }}
+  data-test={{hook (concat hook '-item-container') index=index }}
+>
+  <div class='frost-list-item-container-base'>
+    {{#if (and (or itemExpansion isTypedItemWithExpansion) (not isTypedItemWithoutExpansion))}}
+      {{frost-list-item-expansion
+        hook=(concat hookPrefix '-expansion')
+        hookQualifiers=(hash index=index)
+        model=model.content
+        isExpanded=model.states.isExpanded
+        onExpand=onExpand
+      }}
+    {{else if isAnyTypedItemExpansion}}
+      <div class="item-expansion-placeholder"></div>
+    {{/if}}
+
+    {{#if onSelectionChange}}
+      {{frost-list-item-selection
+        hook=(concat hookPrefix '-selection')
+        hookQualifiers=(hash index=index)
+        model=model.content
+        isSelected=model.states.isSelected
+        onSelect=onSelect
+      }}
+    {{/if}}
+
+    {{component itemComponent
+      hook=(concat hookPrefix '-item')
+      hookQualifiers=(hash index=index)
+      model=model.content
+      isExpanded=model.states.isExpanded
+      isSelected=model.states.isSelected
+      onSelect=onSelect
+    }}
+  </div>
+
+  {{#if model.states.isExpanded}}
+    <div class='frost-list-item-container-expansion'>
+      {{#if isTypedItemWithExpansion}}
+        {{component typedItemExpansionComponent
+          hook=(concat hookPrefix '-itemExpansion')
+          hookQualifiers=(hash index=index)
+          model=model.content
+          isExpanded=model.states.isExpanded
+          isSelected=model.states.isSelected
+          onSelect=onSelect
+        }}
+      {{else if isTypedItemWithoutExpansion}}
+        {{!-- Don't render default expansion template --}}
+      {{else}}
+        {{component itemExpansion
```

Comment:
Let's try to avoid duplicating code.

Reply:
I had previously thought that I couldn't pass `undefined` to a component helper and that was the reason why I thought this duplication was necessary, but I just tried it and there are no errors so I can refactor this logic into a CP to remove the duplication.